plasmapy/examples/plot_magnetic_statics.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+"""
+Magnetostatic Fields
+================
+
+An example of using PlasmaPy's `Magnetostatic` class in `physics` subpackage.
+"""
+
+import plasmapy as pp
+from plasmapy import Plasma
+import numpy as np
+import astropy.units as u
+import matplotlib.pyplot as plt
+import itertools
+
+############################################################
+# Some common magnetostatic fields can be generated and added to a plasma object.
+
+# ## A dipole
+
+dipole = pp.physics.MagneticDipole(np.array([0, 0, 1])*u.A*u.m*u.m, np.array([0, 0, 0])*u.m)
+dipole
+
+############################################################
+# initialize a a plasma, where the magnetic field will be calculated on
+
+plasma = Plasma(domain_x=np.linspace(-2, 2, 30) * u.m,
+                domain_y=np.linspace(0, 0, 1) * u.m,
+                domain_z=np.linspace(-2, 2, 20) * u.m)
+
+############################################################
+# add the dipole field to it
+plasma.add_magneticstatic(dipole)
+
+X, Z = plasma.grid[0, :, 0, :], plasma.grid[2, :, 0, :]
+U = plasma.magnetic_field[0, :, 0, :].value.T  # because grid uses 'ij' indexing
+W = plasma.magnetic_field[2, :, 0, :].value.T  # because grid uses 'ij' indexing
+
+
+############################################################
+plt.figure()
+plt.axis('equal')
+plt.title('Dipole field in x-z plane, generated by a dipole pointing in the z direction')
+plt.streamplot(plasma.x.value, plasma.z.value, U, W)
+
+############################################################
+# ### A circular current-carring wire
```

Comment:
Cut the three #'s here! They're staying in the otherwise beautiful example. I guess you're trying to do a level 3 header? I woner how to do that in autogenerated examples... will check.

Reply:
https://1038-46810954-gh.circle-artifacts.com/0/home/circleci/project/docs/_build/html/auto_examples/plot_magnetic_statics.html#sphx-glr-auto-examples-plot-magnetic-statics-py